### PR TITLE
issue: 1017343 Avoid redundant functions call at lock_wrapper.h

### PR DESCRIPTION
**Pull request by liranoz12**
Signed-off-by: Liran Oz <lirano@mellanox.com>

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1022/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1532/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1533/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1023/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1537/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1024/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1556/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1560/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1561/ for details (Mellanox internal link).

**Comment by OphirMunk** [comment]
Merged #346

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1573/ for details (Mellanox internal link).

File: src/utils/lock_wrapper.h [hunk 1]
```diff
@@ -248,19 +242,15 @@ class lock_spin_recursive : public lock_spin
...
 			m_owner = m_invalid_owner;
 			return lock_spin::unlock();
 		}
 		return 0;
 	};
 	inline int is_locked_by_me() {
-#ifdef DEFINED_NO_THREAD_LOCK
-		return 1;
-#endif // DEFINED_NO_THREAD_LOCK
+		DEFINED_NO_THREAD_LOCK_RETURN
```
Comment (igor-ivanov) [inline src/utils/lock_wrapper.h]: @liranoz12 @OphirMunk in original code this function returns 1 but in new return 0. Is it expected?

File: src/utils/lock_wrapper.h [hunk 2]
```diff
@@ -214,27 +212,23 @@ class lock_spin_recursive : public lock_spin
 	~lock_spin_recursive() {};
 
 	inline int lock() {
-#ifdef DEFINED_NO_THREAD_LOCK
-		return 0;
-#endif // DEFINED_NO_THREAD_LOCK
+		DEFINED_NO_THREAD_LOCK_RETURN
```
Comment (igor-ivanov) [inline src/utils/lock_wrapper.h]: @liranoz12 @OphirMunk is there probability that enabling DEFINED_NO_THREAD_LOCK forces compilation error for example 'Unreachable code'?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1588/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1590/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1591/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1603/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1604/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1612/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1614/ for details (Mellanox internal link).

**Pull request closed without merging**